src/Workspaces/Core/Portable/Storage/SQLite/v2/Interop/SqlStatement.cs

```diff
@@ -67,7 +69,34 @@ public Result Step(bool throwOnError = true)
         }
 
         internal void BindStringParameter(int parameterIndex, string value)
-            => _connection.ThrowIfNotOk(NativeMethods.sqlite3_bind_text(_rawStatement, parameterIndex, value));
+        {
+            // Attempt to stackalloc utf8 converted small strings to avoid lots of allocs.
+            var utf8ByteCount = Encoding.UTF8.GetByteCount(value);
```

Comment:
`GetMaxByteCount` would perform better and may give close enough results to not matter.

Reply:
but we need to know where to place hte 0 to ensure this is encoded properly for sqlite right?  or are we guaranteed that stackalloc'ed memory is zerofilled?

Reply:
I'm not sure on the zero-filled contract. But you get the exact byte count later when you *actually* encode the characters.
As it is now you basically take the perf hit of encoding twice. You can bring that closer to once by getting the "ceiling" length.

Reply:
yeah, looks like it's undefined: https://github.com/dotnet/csharplang/blob/master/spec/unsafe-code.md#stack-allocation

> The content of the newly allocated memory is undefined.

Reply:
Ohhhhhhhhhh.  I'm an idiot.  we can use the return value to figure this out.  tnx.

Reply:
Though i still prefer this approach.  The reason being that using that helper *thirds* the lngths we'd support this up to: https://github.com/dotnet/runtime/blob/1d9e50cb4735df46d3de0cee5791e97295eaf588/src/libraries/System.Private.CoreLib/src/System/Text/UTF8Encoding.cs#L56

That's actually pretty significant here as one of the pieces of data tht comes through here are paths, and i checked roslyn itself to see how large our paths can get (quite large is the answer).  So i'd rather just compute accurately and have this work for almost all cases for the normal 8bit encoding that we'd expect for all chars, versus the conservative 24bit encoding it thinks might be necessary.

Reply:
Fair enough.